src/plugins/ipt/ipt.h

```diff
@@ -0,0 +1,167 @@
+ * If you have any questions about the licensing restrictions on using     *
+ * DRAKVUF in other works, are happy to help.  As mentioned above,         *
+ * alternative license can be requested from the author to integrate       *
+ * DRAKVUF into proprietary applications and appliances.  Please email     *
+ * tamas.k.lengyel@gmail.com for further information.                      *
+ *                                                                         *
+ * If you have received a written license agreement or contract for        *
+ * Covered Software stating terms other than these, you may choose to use  *
+ * and redistribute Covered Software under those terms instead of these.   *
+ *                                                                         *
+ * Source is provided to this software because we believe users have a     *
+ * right to know exactly what a program is going to do before they run it. *
+ * This also allows you to audit the software for security holes.          *
+ *                                                                         *
+ * Source code also allows you to port DRAKVUF to new platforms, fix bugs, *
+ * and add new features.  You are highly encouraged to submit your changes *
+ * on https://github.com/tklengyel/drakvuf, or by other methods.           *
+ * By sending these changes, it is understood (unless you specify          *
+ * otherwise) that you are offering unlimited, non-exclusive right to      *
+ * reuse, modify, and relicense the code.  DRAKVUF will always be          *
+ * available Open Source, but this is important because the inability to   *
+ * relicense code has caused devastating problems for other Free Software  *
+ * projects (such as KDE and NASM).                                        *
+ * To specify special license conditions of your contributions, just say   *
+ * so when you send them.                                                  *
+ *                                                                         *
+ * This program is distributed in the hope that it will be useful, but     *
+ * WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the DRAKVUF   *
+ * license file for more details (it's in a COPYING file included with     *
+ * DRAKVUF, and also available from                                        *
+ * https://github.com/tklengyel/drakvuf/COPYING)                           *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef IPT_H
+#define IPT_H
+
+#include <array>
+#include <fstream>
+
+#include "plugins/private.h"
+#include "plugins/plugins_ex.h"
+
+#define MAX_DRAKVUF_VCPU 16
+
+#define PTW_CURRENT_CR3  (0xC3000000)
+#define PTW_CURRENT_TID  (0x1D000000)
+#define PTW_EVENT_ID     (0xCC000000)
+#define PTW_ERROR_EMPTY  (0xBAD10000)
```

Comment:
Some documentation on these values would be nice

Reply:
This is for custom PTWRITE packets that we inject into IPT data stream in order to add some extra metadata in appropriate places. The values are 100% arbitrary.

Reply:
I know, that's why some documentation on this would be nice ;)

Reply:
Adding an entry to the wiki about ipt and these extra custom packets would be great for example